bindings/nodejs/lib/types/block/parents/index.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { BlockId } from '../id';
+
+type TupMinMax<
+    T,
+    Min extends number,
+    Max extends number,
+    A extends (T | undefined)[] = [],
+    O extends boolean = false,
+> = O extends false
+    ? Min extends A['length']
+        ? TupMinMax<T, Min, Max, A, true>
+        : TupMinMax<T, Min, Max, [...A, T], false>
+    : Max extends A['length']
+    ? A
+    : TupMinMax<T, Min, Max, [...A, T?], false>;
```

Comment:
Is there no easier way? :carlos:

Reply:
Is this ASCII art?

Reply:
I couldnt find one no :D otherwise serialising would be annoying

Reply:
How is this related to serialization?

Reply:
Making fields just array does not ensure length. making them classes would serialize them differently i think.
Anyways this is just the equivalent of `BoxedSlicePrefix<BlockId, BoundedU8<MIN, MAX>`, which i think we will be using more often around nodejs instead of manually verifying at constructor/setter

Reply:
> Making fields just array does not ensure length

Yeah but we should never send an invalid Block from the Rust side and the other way around it will be validated on the Rust side
We also don't validate all the other things

Reply:
Alright ive removed it